### PR TITLE
feat(CI): test currently live world languages

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
This doesn't change the `CLIENT_LOCALE` yet, I just wanted to test changing `CURRICULUM_LOCALE`, but if that works I think we can change both.

EDIT: `CURRICULUM_LOCALE` worked as expected (the Chinese failed), so I've added `CLIENT_LOCALE` to test that.

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
TODO (in separate PRs before merging this)

- [ ] make comment checks handle comments that are substrings of other comments
- [ ] improve client tests and ensure they pass in all languages

**Comment by ojeytonwilliams** [comment]
The root cause of the failing client checks is that the `NavLinks` component depends on `env.json` and the tests don't.  I think this is just a bad pattern and we should use React's Context instead.

**Comment by ojeytonwilliams** [comment]
@nhcarrigan any idea why the final check isn't passing when the rest are?

**Comment by nhcarrigan** [comment]
Ahhhh hold on. Lemme look at the branch protection rules.

**Comment by nhcarrigan** [comment]
image

Yeah, so the change in the matrix triggered a change in the CI names - simple enough to switch this over, but can you confirm that we want all three test suites running on each PR? Need to know which checks are required.

/cc @raisedadead

**Comment by ojeytonwilliams** [comment]
Oh, wait, is it just that the CI expected that because of the old config? And, yep, we want to run it on each PR. At least initially, so we can be confident everything's working as expected.

We might need to change the process if we add too many languages, though.

**Comment by nhcarrigan** [comment]
Yeah the config is looking for the old CI ("old" meaning just before you changed it lol) - I'll flip it to look for the three new ones instead.

**Comment by ojeytonwilliams** [comment]
If I understand this (which I might not), once this PR is in, it'll change the requirements for new PRs, but not old ones.  So, we should be okay, right?

In principle an existing PR could have an issue which the new language tests catch, but those tests would run on merge.  It's not ideal, but we should get warned if there's a problem.

**Comment by nhcarrigan** [comment]
Well, the trick is in the timing for setting the `required` flag - if I switch it now, before this is approved, we end up with PRs that are potentially looking for CI tests that don't exist yet. So they'll be stuck on "Waiting for status" (like this one).

**Review by naomi-lgbt — DISMISSED** [review]
I keep talking about it but I guess I could actually approve it.

**Comment by ojeytonwilliams** [comment]
> Well, the trick is in the timing for setting the `required` flag - if I switch it now, before this is approved, we end up with PRs that are potentially looking for CI tests that don't exist yet. So they'll be stuck on "Waiting for status" (like this one).

Ah, I guess I don't understand. How're you setting this flag?

**Comment by nhcarrigan** [comment]
The setting is on the branch protection configuration.

When I turn on those three as "required status checks", PRs will be blocked from merge until those checks pass.
But since those checks don't technically exist in our repo (yet), they can't pass.

**Comment by ojeytonwilliams** [comment]
Awesome, thanks Nick.  I probably should have known that!  

I think we can use admin powers to bypass that for this PR, then we should be able to set them as required without putting other PRs in an impossible situation.  Let's give Mrugesh a chance weigh in (and review it), though.

**Review by raisedadead — CHANGES_REQUESTED** [review]
I propose we add a third job to the workflow file with the language matrix consisting of [chinese, espanol] and leave the curent job (`test`) as is. That is makes it so that we have the current mandatory check and additional indicative checks in chinese and espanol.

In future, we can consider a dedicated workflow file for testing all languages, that we do not need to run on all PRs. The point being our consistency will grow over time and by the time we are ready for newer languages our coverage of the existing bugs would have increased.

File: .github/workflows/node.js-tests.yml [hunk 1]
```diff
@@ -84,3 +84,53 @@ jobs:
 
       - name: Run Tests
         run: npm test
+
+  test-languages:
+    name: Test Languages
```
Comment (ojeytonwilliams) [inline .github/workflows/node.js-tests.yml:89]: 'Test Languages' isn't a great name, but nothing else leapt to mind

Comment (raisedadead) [inline .github/workflows/node.js-tests.yml:89]:
```suggestion
  test-localization:
    name: Localize
```

How about this?

**Review by raisedadead — APPROVED** [review]

**Review by naomi-lgbt — APPROVED** [review]
Looks good to me.

**Pull request merged**